### PR TITLE
hmr example with runtime and buildtime remotes

### PR DESCRIPTION
**Pull request by knames**
This is a copy from my https://github.com/knames/mf-runtime/tree/main repo, I'm trying to solve an issue with a placeholder `publicPath` and using mf-manifest.json. 

@ScriptedAlchemy said to throw a PR up for everyone.

run `host` and `app2` for build time remotes and HMR

run `runhost` and `app2` for runtime api remotes and HMR.

feel free to change whatever you guys want or if you want me to trim things down, LMK

### CODE REVIEW

File: rspack_hmr/app2/rspack.config.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+console.log({ deps });
+const { ModuleFederationPlugin } = require("@module-federation/enhanced/rspack");
+
+const name = "app_02";
+const name1 = name + "1";
+/**
+ * @type {import('@rspack/cli').Configuration}
+ */
+module.exports = {
+  //context: __dirname,
```
Comment (mich4xD) [inline rspack_hmr/app2/rspack.config.js]: Just wondering why context commented out and type import above too

File: rspack_hmr/app2/rspack.config.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
+const rspack = require("@rspack/core");
+const refreshPlugin = require("@rspack/plugin-react-refresh");
+const isDev = process.env.NODE_ENV === "development";
+
+const path = require("path");
+const deps = require("./package.json").dependencies;
+console.log({ deps });
+const { ModuleFederationPlugin } = require("@module-federation/enhanced/rspack");
+
+const name = "app_02";
```
Comment (mich4xD) [inline rspack_hmr/app2/rspack.config.js:10]: const fileName = ...
const uniqueName = fileName + "1"

As a suggestion. Do you need them to be different?

File: rspack_hmr/app2/rspack.config.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+const name1 = name + "1";
+/**
+ * @type {import('@rspack/cli').Configuration}
```
Comment (mich4xD) [inline rspack_hmr/app2/rspack.config.js:13]: commented code

File: rspack_hmr/runhost/.gitignore [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# Logs
+logs
+*.log
```
Comment (mich4xD) [inline rspack_hmr/runhost/.gitignore:3]: Could get away with a single gitignore at the base of the repo?

File: rspack_hmr/runhost/src/bootstrap.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    beforeInit(args) {
+      console.log("beforeInit: ", args);
+      return args;
+    },
+    beforeRequest(args) {
+      console.log("beforeRequest: ", args);
+      return args;
+    },
+
+    // loadRemoteSnapshot(args) {
```
Comment (mich4xD) [inline rspack_hmr/runhost/src/bootstrap.tsx:19]: Commented out

File: rspack_hmr/runhost/src/bootstrap.tsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    },
+    async loadShare(args) {
+      console.log("loadShare:", args);
+      return args;
+    },
+    async beforeLoadShare(args) {
+      console.log("beforeloadShare:", args);
+      return args;
+    },
+    // async initContainer(args) {
```
Comment (mich4xD) [inline rspack_hmr/runhost/src/bootstrap.tsx:52]: Commented out

File: rspack_hmr/app2/rspack.config.js (hunk 3 above)
Comment (ScriptedAlchemy) [reply rspack_hmr/app2/rspack.config.js:13]: its type import for jsdoc not comment

File: rspack_hmr/app2/rspack.config.js [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    new ModuleFederationPlugin({
+      name: name,
+      filename: "remoteEntry.js",
+      exposes: {
+        "./Hello": "./src/Hello.tsx",
+        "./pi": "./src/pi.ts",
+      },
+      manifest: true,
+      // shared: {
+      //   ...deps,
```
Comment (ScriptedAlchemy) [inline rspack_hmr/app2/rspack.config.js:107]: @knames why is module sharing disabled?

Comment (knames) [reply rspack_hmr/app2/rspack.config.js:107]: @ScriptedAlchemy sorry the code was in a less than polished state, i only whipped it up to tinker with runtime without all the crazy work code.  it should work fine if you share react and react-dom, i'm not sure that `...deps` will work though

File: rspack_hmr/app2/rspack.config.js (hunk 2 above)
Comment (knames) [reply rspack_hmr/app2/rspack.config.js:10]: i thiiiink that it needs a unique name for hmr to work properly. +1 isn't ideal, but simple for a sample project

**Pull request merged**